spec-dtslint/Subject-spec.ts

```diff
@@ -0,0 +1,39 @@
+import { Subject } from 'rxjs';
+
+it('should have the proper types for Subject<number>', () => {
+  const subject = new Subject<number>(); // $ExpectType Subject<number>
```

Comment:
Constructors require a generic type if there are no arguments for you to infer from (such as this case).  Otherwise you'd get the default type of `unknown` all the time.

Reply:
> Does this syntax go against the advice we've given in the past of not specifying generic args, or are constructors the exception to this rule?

Constructors are essentially types, so the usage is fine. It should be avoided only where the explicit type parameter is redundant.

The places where explicit type parameters should be always be avoided are for things that are not in the value space rather than the type space - e.g. for function calls. Explicitly specifying type parameters for function/methods calls to type the passed arguments is the big no-no.